File: csharp/Types/EventIssue.cs
using System.Text.Json.Serialization;

namespace Royaltics.ErrorTracker.Types;

public sealed record EventContext
{
    [JsonPropertyName("culprit")]
    public required string Culprit { get; init; }
    
    [JsonPropertyName("extra")]
    public Dictionary<string, string>? Extra { get; init; }
    
    [JsonPropertyName("platform")]
    public string? Platform { get; init; }
    
    [JsonPropertyName("app")]
    public string? App { get; init; }
    
    [JsonPropertyName("version")]
    public string? Version { get; init; }
    
    [JsonPropertyName("device")]
    public string? Device { get; init; }
    
    [JsonPropertyName("tags")]
    public List<string>? Tags { get; init; }
}

public sealed record SerializedError
{
    [JsonPropertyName("name")]
    public required string Name { get; init; }
    
    [JsonPropertyName("message")]
    public required string Message { get; init; }
    
    [JsonPropertyName("stack")]
    public string? Stack { get; init; }
    
    [JsonPropertyName("extra")]
    public Dictionary<string, string>? Extra { get; init; }
}

public sealed record EventIssue
{
    [JsonPropertyName("event_id")]
    public required string EventId { get; init; }
    
    [JsonPropertyName("title")]
    public required string Title { get; init; }
    
    [JsonPropertyName("level")]
    public required string Level { get; init; }
    
    [JsonPropertyName("event")]
    public required SerializedError Event { get; init; }
    
    [JsonPropertyName("context")]
    public required EventContext Context { get; init; }
    
    [JsonPropertyName("timestamp")]
    public required string Timestamp { get; init; }
}

public sealed record TransportPayload
{
    [JsonPropertyName("event")]
    public required string Event { get; init; }
    
    [JsonPropertyName("license_id")]
    public required string LicenseId { get; init; }
    
    [JsonPropertyName("license_name")]
    public string? LicenseName { get; init; }
    
    [JsonPropertyName("license_device")]
    public required string LicenseDevice { get; init; }
}

File: php/src/Types/EventIssue.php
<?php

declare(strict_types=1);

namespace Royaltics\ErrorTracker\Types;

final readonly class EventContext
{
    public function __construct(
        public string $culprit,
        public ?array $extra = null,
        public ?string $platform = null,
        public ?string $app = null,
        public ?string $version = null,
        public ?string $device = null,
        public ?array $tags = null
    ) {}
}

final readonly class SerializedError
{
    public function __construct(
        public string $name,
        public string $message,
        public ?string $stack = null,
        public ?array $extra = null
    ) {}
}

final readonly class EventIssue
{
    public function __construct(
        public string $eventId,
        public string $title,
        public string $level,
        public SerializedError $event,
        public EventContext $context,
        public string $timestamp
    ) {}
}

final readonly class TransportPayload
{
    public function __construct(
        public string $event,
        public string $licenseId,
        public string $licenseDevice,
        public ?string $licenseName = null
    ) {}
}
